lennonFolder/lennon-github-serach/simple-github/src/main/kotlin/com/androidhuman/example/simplegithub/ui/main/MainActivity.kt

```diff
@@ -0,0 +1,20 @@
+package com.androidhuman.example.simplegithub.ui.main
+
+import android.content.Intent
+import android.os.Bundle
+import android.support.design.widget.FloatingActionButton
+import android.support.v7.app.AppCompatActivity
+import android.view.View
+import com.androidhuman.example.simplegithub.R
+import com.androidhuman.example.simplegithub.ui.search.SearchActivity
+
+// 이 액티비티는 현재 저장소 검색 액티비티를 호출하는 기능만 구현
+class MainActivity : AppCompatActivity() {
+    internal lateinit var btnSearch: FloatingActionButton
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        btnSearch = findViewById(R.id.btnActivityMainSearch)
+        btnSearch.setOnClickListener(View.OnClickListener { startActivity(Intent(this@MainActivity, SearchActivity::class.java)) })
```

Comment:
아래와 같이 수정하면 가독성이 더 좋을거 같아 제안드립니다~
```suggestion
btnSearch.setOnClickListener{ goToSearch() }

fun goToSearch() =
​startActivity(Intent(this, SearchActivity::class.java))
```

Reply:
startActivity 함수는 Activity에 있는 함수로 goToSearch함수에 패러미터로 Activity를 넘겨줘야 할 것 같습니다. 그래서 개인적으로 코드가 좀 길어보이는데, `        btnSearch.setOnClickListener {
            startActivity(Intent(this, SearchActivity::class.java)) }`로 수정하는건 어떨까요?

Reply:
goToSearch 가 Activity 안에서만 실행되서 파라미터로 안넘겨줘도 됩니다~

레논말처럼 한줄로 표현되는 코드이기에 별도의 함수로 안빼도 되는데 저는 한줄이라도 함수 네이밍으로 해당 코드의 목적, 행동을 명시하는걸 선호해서 제안 드렸습니다~.

Reply:
goToSearch()가  Activity에서 실행되므로 그냥 startActivity가 호출이 되군요. 

제가 생각하는 함수를 만드는 철학은 `재활용이 되는가?`였습니다. 한곳에서만 쓰이는 코드이거나 어떻게 돌아가는지 한눈에 보이는 코드는 굳이 함수를 만들지 말자는 게 제 코딩 철학? 이 었는데, 휴 말씀처럼 한 번만 쓰이는 코드라도 협업하는 입장에서 명시적으로 어떤 동작을 하는지 함수를 만들어 작성하는 것도 좋아 보이네요.

Reply:
네네 딱히 어떤걸로 해도 상관없는 취향의 영역이긴 합니다.